packages/hierarchy-builder/src/test/internal/operators/Grouping.test.ts

```diff
@@ -0,0 +1,207 @@
+                grouped: [
+                  {
+                    label: "TestSchema A",
+                    key: {
+                      type: "class-grouping",
+                      class: {
+                        name: "TestSchema A",
+                      },
+                    },
+                    children: [allNodes[0]],
+                  },
+                ],
+                ungrouped: [allNodes[1]],
+                groupingType: "class",
+              } as GroupingHandlerResult;
+            },
+            async (allNodes) => {
+              return {
+                grouped: [
+                  {
+                    label: "1",
+                    key: {
+                      type: "label-grouping",
+                      label: "1",
+                    },
+                    children: allNodes.filter((node) => !HierarchyNode.isClassGroupingNode(node)),
+                  },
+                ],
+                ungrouped: allNodes.filter((node) => HierarchyNode.isClassGroupingNode(node)),
+                groupingType: "label",
+              } as GroupingHandlerResult;
+            },
+          ]),
+        ),
+      );
+      expect(applyGroupingHidingParamsStub.callCount).to.eq(3);
+      expect(result).to.deep.eq([
+        {
+          label: "1",
+          key: {
+            type: "label-grouping",
+            label: "1",
+          },
+          children: [nodes[1]],
+        },
+        {
+          label: "TestSchema A",
+          key: {
+            type: "class-grouping",
+            class: {
+              name: "TestSchema A",
+            },
+          },
+          children: [
+            {
+              label: "1",
+              key: {
+                type: "label-grouping",
+                label: "1",
+              },
+              children: [nodes[0]],
+            },
+          ],
+        },
+      ]);
+    });
+  });
+
+  describe("createGroupingHandlers", () => {
+    let createBaseClassGroupingHandlersStub: sinon.SinonStub;
+    let createClassGroupsStub: sinon.SinonStub;
+    let createLabelGroupsStub: sinon.SinonStub;
+    before(() => {
+      createBaseClassGroupingHandlersStub = sinon.stub(baseClassGrouping, "createBaseClassGroupingHandlers").resolves([] as GroupingHandler[]);
+      createClassGroupsStub = sinon.stub(classGrouping, "createClassGroups").resolves({ grouped: [], ungrouped: [], groupingType: "class" });
+      createLabelGroupsStub = sinon.stub(labelGrouping, "createLabelGroups").resolves({ grouped: [], ungrouped: [], groupingType: "label" });
+    });
+
+    it("creates grouping handlers in class -> label grouping order", async () => {
+      const result = await createGroupingHandlers(metadata, []);
+      expect(createBaseClassGroupingHandlersStub.callCount).to.eq(1);
+      expect(result.length).to.eq(2);
+      expect(createClassGroupsStub.callCount).to.eq(0);
+      await result[0]([]);
+      expect(createClassGroupsStub.callCount).to.eq(1);
```

Comment:
would an `expect(result[0]).to.eq(createClassGroupsStub)` check work here?

Reply:
No it would not, result[i] returns a function that calls the stub, not the stub itself:
![image](https://github.com/iTwin/presentation/assets/100586436/03035d91-5182-46ce-b217-c32e78aba93b)